### PR TITLE
Alec - Added Create Students page

### LINKED ISSUE #47: SUBFEATURE: Create page for Students plus tests and stories

### PR DESCRIPTION
**Pull request by alecsong222**
## Overview

This PR adds the Create Students frontend page to interface with the backend Create functionality.
Branch - https://github.com/ucsb-cs156-f24/proj-happycows-f24-09/tree/Alec-StudentsCreatePage

## Screenshots (Optional)

image

## Validation (Optional)

Go to https://happycows-dev-alecsong222-students.dokku-09.cs.ucsb.edu and login
Use Swagger to create a course in the backend.
Go back to the homepage and click the dropdown menu for Admins and select students
Try creating a student with no fields filled; ensure the proper error messages are thrown
image
Fill out the fields and create a student; ensure that the correct message shows up. Use the corresponding id to the course you just created to make the student, not the name of the course.
image
Go back to the index page and ensure that you can view your student, and all fields are filled out correctly

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #47 

## PR Checklist
- [x] Commented code removed
- [x] PR deals with only one concern
- [x] PR Title is descriptive enough
- [x] PR description details what a user/dev will notice is different
- [x] No merge conflicts
- [x] Deployed and linked with Dokku deployment
- [x] New branch updated with main
- [x] PR is linked to a branch
- [x] PR is linked to an issue/ticket
- [x] Linked issue is in "In Review" column on Kanban board
- [x] PR is tagged with the Team's tag
- [x] All test cases pass

### CODE REVIEW

**Review by TomShangguan — DISMISSED** [review]
LGTM! I tried to create a student follow your steps and it works!
Screenshot 2024-11-25 at 12 02 13

**Review by elijahelephant — DISMISSED** [review]
I tested everything on dokku. Code looks decent as far as I can tell but I am not an expert in JS. Approved.

Edit: Looks like you should pull main first before asking for staff, let me know if you need help.

**Review by pconrad — CHANGES_REQUESTED** [review]
Similar feedback to code review on PRs #54 and #57

**Review by pconrad — DISMISSED** [review]
https://github.com/ucsb-cs156-f24/proj-happycows-f24-09/pull/43/files should not have been merged without adding validation for the courseId.

Something like this:
```
        coursesRepository.findById(courseId).orElseThrow(() -> new EntityNotFoundException(Courses.class, courseId));
```

I won't revoke the points you already earned.  But I also won't merge anything else that touches CRUD operations for Student (front end or backend) unless until this fix is part of that PR.

File: frontend/src/main/components/Students/StudentsEditForm.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        { defaultValues: initialContents || {}, }
+    );
+    // Stryker restore all
+
+    const navigate = useNavigate();
+
+    const testIdPrefix = "StudentsEditForm";
+
+    // For explanation, see: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime
+    // Note that even this complex regex may still need some tweaks
```
Comment (pconrad) [inline frontend/src/main/components/Students/StudentsEditForm.js]: See code review on #57

**Review by pconrad — DISMISSED** [review]
Same issue as on the previous PR

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**